daemon/content.go

```diff
@@ -26,5 +30,126 @@ func (daemon *Daemon) configureLocalContentStore() (content.Store, leases.Manage
 	}
 	md := metadata.NewDB(db, cs, nil)
 	daemon.mdDB = db
-	return md.ContentStore(), metadata.NewLeaseManager(md), nil
+	return namespacedContentProvider(md.ContentStore(), ns), namespacedLeaseManager(metadata.NewLeaseManager(md), ns), nil
+}
+
+func withNamespace(ctx context.Context, namespace string) context.Context {
+	ns, ok := namespaces.Namespace(ctx)
+	if !ok {
+		ns = namespace
+	}
+	return namespaces.WithNamespace(ctx, ns)
```

Comment:
Isn't it redundant to create a child context with the same namespace value as its parent context?
```suggestion
	if _, ok := namespaces.Namespace(ctx); ok {
		return ctx
	}
	return namespaces.WithNamespace(ctx, namespace)
```